pkg/pod/status.go

```diff
@@ -217,16 +219,30 @@ func createMessageFromResults(results []v1beta1.PipelineResourceResult) (string,
 	return string(bytes), nil
 }
 
-func filterResultsAndResources(results []v1beta1.PipelineResourceResult) ([]v1beta1.TaskRunResult, []v1beta1.PipelineResourceResult, []v1beta1.PipelineResourceResult) {
+func filterResultsAndResources(ctx context.Context, results []v1beta1.PipelineResourceResult) ([]v1beta1.TaskRunResult, []v1beta1.PipelineResourceResult, []v1beta1.PipelineResourceResult) {
 	var taskResults []v1beta1.TaskRunResult
 	var pipelineResourceResults []v1beta1.PipelineResourceResult
 	var filteredResults []v1beta1.PipelineResourceResult
 	for _, r := range results {
 		switch r.ResultType {
 		case v1beta1.TaskRunResultType:
+			cfg := config.FromContextOrDefaults(ctx)
+			aos := v1beta1.ArrayOrString{}
+			if cfg.FeatureFlags.EnableAPIFields == config.AlphaAPIFields {
+				// Note that for unsupported types such as []int will be Unmarshalled to String
+				err := aos.UnmarshalJSON([]byte(r.Value))
+				if err != nil {
```

Comment:
this if statement will not be reached at all, right? since `UnmarshalJSON` does not return any error.

Reply:
Thanks for pointing that out! 
Yes right now we won't reach this, but considering that `UnmarshalJSON` may be changed (maybe one day we need to return err for some cases and don't have to convert them to string), maybe it is ok to keep it?